go/protocol/keybase1/block.go

```diff
@@ -113,6 +113,22 @@ func (o PutBlockArg) DeepCopy() PutBlockArg {
 	}
 }
 
+type PutBlockAgainArg struct {
+	Folder   string         `codec:"folder" json:"folder"`
+	Ref      BlockReference `codec:"ref" json:"ref"`
```

Comment:
@jzila: the behavior of the two RPCs is very different, and yeah the blockref is needed.  This RPC covers the case where a foreground write created a dedup block with a new refnonce, but by the time the journal tries to call `AddReference` the block has been archived/deleted or needs to be put again.  But it needs to use the original reference, since the client will already have put that into directory entries, etc.